rs/crypto/tree_hash/src/lib.rs

```diff
@@ -1307,3 +1307,27 @@ pub trait HashTreeBuilder {
     /// Does not `panic!`.
     fn witness_generator(&self) -> Option<Self::WitnessGenerator>;
 }
+
+/// A restricted match pattern that either allows a label or a wildcard with a list of exceptions.
+#[derive(Clone, Debug)]
+pub enum MatchPattern {
+    Label(Label),
+    Wildcard(Vec<Label>),
```

Comment:
nit: maybe something like this would be clearer to read on the clients side?
```suggestion
    AllLabelsExcept(Vec<Label>),
```